editing/edit-with-branch-versioning/src/main/java/com/esri/samples/edit_with_branch_versioning/EditWithBranchVersioningController.java

```diff
@@ -0,0 +1,356 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.esri.samples.edit_with_branch_versioning;
+
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import javafx.fxml.FXML;
+import javafx.geometry.Point2D;
+import javafx.scene.control.Alert;
+import javafx.scene.control.Button;
+import javafx.scene.control.ComboBox;
+import javafx.scene.control.Label;
+import javafx.scene.control.ProgressIndicator;
+import javafx.scene.control.TextField;
+import javafx.scene.input.MouseButton;
+import javafx.scene.layout.VBox;
+
+import com.esri.arcgisruntime.arcgisservices.ServiceVersionInfo;
+import com.esri.arcgisruntime.arcgisservices.ServiceVersionParameters;
+import com.esri.arcgisruntime.arcgisservices.VersionAccess;
+import com.esri.arcgisruntime.concurrent.ListenableFuture;
+import com.esri.arcgisruntime.data.ArcGISFeature;
+import com.esri.arcgisruntime.data.FeatureTableEditResult;
+import com.esri.arcgisruntime.data.ServiceFeatureTable;
+import com.esri.arcgisruntime.data.ServiceGeodatabase;
+import com.esri.arcgisruntime.geometry.Point;
+import com.esri.arcgisruntime.layers.FeatureLayer;
+import com.esri.arcgisruntime.loadable.LoadStatus;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.GeoElement;
+import com.esri.arcgisruntime.mapping.Viewpoint;
+import com.esri.arcgisruntime.mapping.view.IdentifyLayerResult;
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.security.AuthenticationManager;
+import com.esri.arcgisruntime.security.DefaultAuthenticationChallengeHandler;
+
+public class EditWithBranchVersioningController {
+
+  @FXML private Button createVersionButton;
+  @FXML private Button switchVersionButton;
+  @FXML private ComboBox<String> damageTypeComboBox;
+  @FXML private ComboBox<VersionAccess> accessComboBox;
+  @FXML private Label currentVersionLabel;
+  @FXML private MapView mapView;
+  @FXML private ProgressIndicator progressIndicator;
+  @FXML private TextField descriptionTextField;
+  @FXML private TextField nameTextField;
+  @FXML private VBox createVersionVBox;
+  @FXML private VBox editFeatureVBox;
+
+  private ArcGISFeature selectedFeature;
+  private FeatureLayer featureLayer;
+  private ServiceFeatureTable serviceFeatureTable;
+  private ServiceGeodatabase serviceGeodatabase; // keeps loadable in scope to avoid garbage collection
+  private String defaultVersion;
+  private String userCreatedVersion;
+
+  public void initialize() {
+    try {
+      // create a map with the streets vector basemap
+      ArcGISMap map = new ArcGISMap(Basemap.createStreetsVector());
+
+      // create a map view and set its map
+      mapView.setMap(map);
```

Comment:
Perhaps on more 'advanced' samples, we could consolidate these two lines? The thought would be that users who look at this sample already understand how maps and MapViews work, so they wouldn't need an in-depth explanation of every step?

```suggestion
      // create a map and set it to the map view
      ArcGISMap map = new ArcGISMap(Basemap.createStreetsVector());
      mapView.setMap(map);
```